jobs/integr8ly/browser-based-testsuite-pipeline.yaml

```diff
@@ -48,17 +52,32 @@
                             # Disable starting Selenium, Docker is used instead
                             sed -i 's/"start_process" : true,/"start_process" : false,/g' nightwatch.json
                             npm install
-                            npm test
                             """
+                            
+                            String output = sh returnStdout: true, script: 'npm test || true'
+                            
+                            // To see the output in Jenkins build console log as well
+                            println output
+
+                            if(!output.contains('Enmasse Console Login')) {
+                                currentBuild.result = 'FAILURE'
+                                publishTestResults = false
```

Comment:
What is the purpose of this condition branch?

Reply:
to check if the tests were executed

Reply:
Do we care if they were not executed or they failed?
I just think we could simplify this by not caring.
E.g:
```
try {
  sh 'npm test'
} catch (Exception e) {
  currentBuild.result = 'FAILURE'
}
```
With this^ we won't need any conditions here, nor in the reporting.

Reply:
need to try it out. Will create some broken branch and see if the report is generated. Can change it then

Reply:
@matskiv the `npm tests` exits with 1 in case of test failure. I wanted to ensure that the build is marked as UNSTABLE for test failures and as FAILURE for errors during running tests. In case of such errors the report is not generated at all, so I thought I would skip the reporting stage.

Reply:
@trepel but the solution that achieves that isn't very elegant IMHO. But this are just my 2cents, feel free to ignore.

Reply:
@matskiv no, it is not indeed. It is just a solution I was able to come up with. I don't want to ignore, I would be interested in how to do it better.

Reply:
One alternative that comes to my mind: in the catch block, check if there are any files in reports directory - if yes = unstable, if no = failure.

Reply:
@matskiv I was thinking about your latest suggestion. This would introduce some more code and as a result would not be much cleaner solution. You would need to get the number of entries in the **reports** directory (`ls | wc -l`), then trim the output (unfortunately the output is padded with empty spaces) and check if it is not equal to 0. Do you think it is worth the hassle?

Reply:
I think that's cleaner than stealing stdout, printing it and checking `output.contains('Enmasse Console Login')`

Reply:
OK. I will do it this way. You are probably right that using wc ... is easier to understand